### PR TITLE
fix: dynamically load blocks and videojs

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

### <samp>🤖 Generated by Copilot at 2c50c11</samp>

This pull request enhances the `BlockRenderer` component to dynamically load and render `Video` blocks from the `block` library. This reduces the initial loading time and improves the responsiveness of the journeys UI.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/33281894/todos/6317918540)

# How should this PR be QA Tested?

- [ ] it should dynamically load blocks in the BlockRenderer component, payload size should be reduced
- [ ] it should retain current block functionalities

# Walkthrough

### <samp>🤖 Generated by Copilot at 2c50c11</samp>

*  Dynamically load `Video` component using `next/dynamic` to improve performance and reduce bundle size ([link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71R2), [link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71L11-R14), [link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71R29-R36), [link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71L150-R177))
*  Use `useBlocks` hook from `block` library to access active block and its children in `BlockRenderer.tsx` ([link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71L11-R14), [link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71R81-R89))
*  Conditionally render video block based on the presence of cover block or video block as children of card block ([link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71R81-R89), [link](https://github.com/JesusFilm/core/pull/1683/files?diff=unified&w=0#diff-eecf5b31a3066948b3c27f02cbf8e50ebe0f0b321c21e2ecf0c2d0f37bf1ca71L150-R177))

### CODE REVIEW

File: libs/journeys/ui/src/components/BlockRenderer/BlockRenderer.spec.tsx [hunk 1]
```diff
@@ -415,15 +431,41 @@ describe('BlockRenderer', () => {
         }
       ]
     }
+    const result = jest.fn(() => ({
```
Comment (csiyang) [inline libs/journeys/ui/src/components/BlockRenderer/BlockRenderer.spec.tsx:434]: Can you extract this out into the `describe` level to re use it?

File: libs/journeys/ui/src/components/BlockRenderer/BlockRenderer.spec.tsx [hunk 2]
```diff
@@ -415,15 +431,41 @@ describe('BlockRenderer', () => {
...
+          id: 'uuid',
+          __typename: 'StepViewEvent'
+        }
+      }
+    }))
     const { getByText } = render(
-      <MockedProvider>
+      <MockedProvider
+        mocks={[
+          {
```
Comment (csiyang) [inline libs/journeys/ui/src/components/BlockRenderer/BlockRenderer.spec.tsx:445]: Also extract out to `describe` level

**Review by csiyang — DISMISSED** [review]

**Review by csiyang — APPROVED** [review]

File: apps/journeys/src/components/EmbeddedPreview/EmbeddedPreview.tsx [hunk 3]
```diff
@@ -14,7 +14,6 @@ import useFullscreenStatus from '../../libs/useFullscreenStatus/useFullscreenSta
 import { Conductor } from '../Conductor'
 
 import { ButtonWrapper } from './ButtonWrapper/ButtonWrapper'
-import { VideoWrapper } from './VideoWrapper/VideoWrapper'
```
Comment (GeronimoJohn) [inline apps/journeys/src/components/EmbeddedPreview/EmbeddedPreview.tsx:17]: Removed the use of VideoWrapper for a few reasons: 

- The initial intention for it was to pause the Video Block, if first in journey. 
      - Though it seems that we're just allowing EmbeddedJourneys to play videos
      - Need to clarify if we still want that pause behaviour
- VideoWrapper is outdated and needs to be updated to be supporting YouTube and Cloudflare (needs to be done in another ticket)

Comment (GeronimoJohn) [reply apps/journeys/src/components/EmbeddedPreview/EmbeddedPreview.tsx:17]: @tataihono

**Review by csiyang — APPROVED** [review]

**Review by csiyang — APPROVED** [review]

**Pull request merged**